Fix markdown format of CONTRIBUTING.md (#3173)
Clinical-Genomics/cg: CONTRIBUTING.md

* **Describe the current behavior** and **explain which behavior you expected
  to see instead** and why.
* **Explain why this enhancement would be useful**

#### Local Development

**NEVER USE PREINSTALLED PYTHON**

First of all, make sure that you are managing your python versions that are
used on your machine, [never use the OS native python](https://docs.python-guide.org/starting/installation/).
Suggested ways to handle python version are either through
[homebrew](https://brew.sh/#install)(OSX), [pyenv](https://github.com/pyenv/pyenv) or [conda](https://docs.anaconda.com/anaconda/install/).

For local development, it is recommended to use Poetry. Ensure that you have
Poetry installed and run
```bash
poetry install
```

On our servers where the production and stage versions of CG are run the
packages are maintained by using conda environments. For local development it
is suggested to follow the [python packaging
guidelines](https://packaging.python.org/tutorials/managing-dependencies/)
where it is suggested to manage your local python environment with

> [pipenv](https://pipenv.pypa.io/en/latest/). CG has a Pipfile.lock file which

Author: Yes! Just did. I don't know about the detailed steps for combining conda and poetry, so just removed that, thinking we can add it later if someone wants to take the time?

> [pipenv](https://pipenv.pypa.io/en/latest/). CG has a Pipfile.lock file which

Author: Do you think the changes here (dde34a6adb4c0160e1fdfb7bc3ce9020846e3ae8) are good enough?

#### Genotype (gt)

Interface to Genotype. For uploading results from the workflow about genotypes
to compare and validate that we are clear of sample mix-ups.

#### Housekeeper (hk)

Interface to Housekeeper. For storing files from analysis runs and FASTQ files
from demultiplexing.

#### Loqus

Interface to LoqusDB. For loading observation counts from the analysis output.

#### Osticket

Internal app for opening tickets in SupportSystems. We use this mainly to link
a ticket with the opening of an order for new samples/analyses.

#### Scout (scoutapi)

Interface to Scout. For uploading analysis results to Scout. It's also used to
access the generation of gene panels files used in the analysis workflow.

#### Delivery report

Module to generate Delivery Reports. This module is designed to convey the
results of genetic analysis to the customer. It includes information on sample
characteristics, laboratory preparation, sequencing attributes, as well as data
analysis performance and limitations.

### Cli

> The command line code is written in the [Click](https://click.palletsprojects.com/en/8.1.x/) framework.

Author: Ouch, good catch!

Includes: `stats`, `hk`, `status`

> The API accepts the name of a flowcell which will be looked up in `stats`. For

Reviewer: ```suggestion
The API accepts the name of a flow cell which will be looked up in `stats`. For
```

> all samples on the flowcell it will:

Reviewer: ```suggestion
all samples on the flow cell it will:
```

1. Check if the quality (Q30) is good enough to include the sequencing results
2. update the number of reads that the sample has been sequenced _overall_ and
   match this with the requirement given by the application.
3. accordingly, the interface will look up FASTQ files and store them using `hk`.
4. if a sample has sufficient number of reads, the `sequenced_at` date will be
   filled in (`status`) according to the sequencing date of the most recent
   flowcell.

> Another common task is to transfer data and FASTQ files from the demux/cgstats
> interface when a demultiplexing task completes. This is as easy as determining
> the flowcell of interest and running:
> 
> ```bash
> cg transfer flowcell HGF2KBCXX
> ```

Reviewer: ```suggestion
```

> The command will update the _total_ read counts of each sample and check
> against the application for the sample if it has been fully sequenced. It will
> also make sure to link the related FASTQ files to Housekeeper. You can run the
> command over and over - only additional information will be updated.

Reviewer: ```suggestion
```

Includes: `status`, `hk`, `coverage`

> The API will fetch information about an analysis like name and family ID and

Reviewer: ```suggestion
The API will fetch information about an analysis like name and case ID and
```

related samples from `status`. It will get the Sambamba output from `hk` and
use the `coverage` app interface to upload the data to Chanjo for each
individual sample.

##### Gt

Includes: `status`, `hk`, `tb`, `gt`

Given an analysis, the API will fetch information about the family. It will
fetch the gBCF + qcmetrics files from `hk`. It will parse the qcmetrics file
using `tb` to find out the predicted sex of each sample. It will then upload
the results to Genotype. Subsequent upload of the same samples will overwrite
existing information while logging the event.

##### Observations

Includes: `status`, `hk`, `loqus`

Given an analysis record, it will fetch required files from `hk` and upload the
variants to the observation cound database using `loqus`. This only works for

> families with at least one affected individual.

Reviewer: ```suggestion
cases with at least one affected individual.
```
